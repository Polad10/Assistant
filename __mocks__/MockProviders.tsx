import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
