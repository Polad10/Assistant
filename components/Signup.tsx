import { useContext, useState } from 'react'
import MainView from './MainView'
import { Button } from '@rneui/themed'
import { ThemeContext } from '../contexts/ThemeContext'
import MyInput from './MyInput'
import { createUserWithEmailAndPassword, getAuth } from 'firebase/auth'
import { showDangerMessage } from '../helpers/ToastHelper'
import Toast from 'react-native-root-toast'
import { Api } from '../helpers/Api'
import LoadingView from './LoadingView'
import { Keyboard } from 'react-native'
import { LocalizationContext } from '../contexts/LocalizationContext'

export default function Signup() {
  const themeContext = useContext(ThemeContext)!
  const localizationContext = useContext(LocalizationContext)!
  const auth = getAuth()

  const translator = localizationContext.translator

  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [loading, setLoading] = useState(false)

  const [showEmailInputError, setShowEmailInputError] = useState(false)
  const [showPassowrdInputError, setShowPasswordInputError] = useState(false)

  async function signUp() {
    if (validate()) {
      try {
        setLoading(true)
        Keyboard.dismiss()

        const userCredential = await createUserWithEmailAndPassword(auth, email, password)

        const api = new Api(userCredential.user)
        await api.signUp()
      } catch (error: any) {
        switch (error.code) {
          case 'auth/invalid-email':
            setShowEmailInputError(true)
            showDangerMessage(translator.translate('emailIsInvalid'), Toast.positions.TOP)
            break
          case 'auth/weak-password':
            setShowPasswordInputError(true)
            showDangerMessage(translator.translate('enterPassword'), Toast.positions.TOP)
            break
          case 'auth/email-already-in-use':
            setShowEmailInputError(true)
            showDangerMessage(translator.translate('emailAlreadyInUseMessage'), Toast.positions.TOP)
            break
          default:
            showDangerMessage(translator.translate('somethingWentWrongMessage'), Toast.positions.TOP)
        }
      } finally {
        setLoading(false)
      }
    } else {
      showDangerMessage(translator.translate('fillInAllRequiredFields'), Toast.positions.TOP)
    }
  }

  function validate() {
    let valid = true

    if (!email) {
      setShowEmailInputError(true)
      valid = false
    }

    if (!password) {
      setShowPasswordInputError(true)
      valid = false
    }

    return valid
  }

  function handleEmailChange(val: string) {
    setShowEmailInputError(false)
    setEmail(val)
  }

  function handlePasswordChange(val: string) {
    setShowPasswordInputError(false)
    setPassword(val)
  }

  return (
    <MainView style={{ paddingTop: 20, paddingHorizontal: 10 }}>
      <MyInput
        label={translator.translate('email')}
        placeholder={translator.translate('enterEmail')}
        onChangeText={handleEmailChange}
        showError={showEmailInputError}
        keyboardType='email-address'
        autoCapitalize='none'
      />
      <MyInput
        label={translator.translate('password')}
        placeholder={translator.translate('enterPassword')}
        secureTextEntry={true}
        onChangeText={handlePasswordChange}
        showError={showPassowrdInputError}
      />
      <Button
        title={translator.translate('signUp')}
        size='lg'
        buttonStyle={{ borderRadius: 10 }}
        color={themeContext.accent}
        onPress={signUp}
      />
      {loading && <LoadingView />}
    </MainView>
  )
}
