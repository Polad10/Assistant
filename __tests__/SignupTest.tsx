import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import Signup from '../components/Signup'

describe('Signup', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <Signup />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
