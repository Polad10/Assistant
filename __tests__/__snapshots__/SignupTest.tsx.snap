// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#121212",
        "flex": 1,
      },
      {
        "paddingHorizontal": 10,
        "paddingTop": 20,
      },
    ]
  }
>
  <View
    style={
      {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
    testID="RNE__Input__view-wrapper"
  >
    <Text
      style={
        {
          "color": "#d3fbd8",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Email
    </Text>
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#374151",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 20,
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        editable={true}
        keyboardType="email-address"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="your-email@example.com"
        placeholderTextColor="#86939e"
        returnKeyType="done"
        selectionColor="#faf8ff"
        showError={false}
        style={
          {
            "color": "#faf8ff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
            "paddingHorizontal": 10,
            "paddingTop": 20,
            "paddingVertical": 20,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        {
          "color": "#ff190c",
          "fontSize": 12,
          "height": 0,
          "margin": 0,
          "padding": 0,
        }
      }
    />
  </View>
  <View
    style={
      {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
    testID="RNE__Input__view-wrapper"
  >
    <Text
      style={
        {
          "color": "#d3fbd8",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Password
    </Text>
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#374151",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 20,
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        autoCorrect={false}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Must have at least 6 characters"
        placeholderTextColor="#86939e"
        returnKeyType="done"
        secureTextEntry={true}
        selectionColor="#faf8ff"
        showError={false}
        style={
          {
            "color": "#faf8ff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
            "paddingHorizontal": 10,
            "paddingTop": 20,
            "paddingVertical": 20,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        {
          "color": "#ff190c",
          "fontSize": 12,
          "height": 0,
          "margin": 0,
          "padding": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 2,
        },
        undefined,
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": false,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#7600eb",
            "borderColor": "#2089dc",
            "borderRadius": 10,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 12,
            "paddingHorizontal": 14,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Sign up
        </Text>
      </View>
    </View>
  </View>
</View>
`;
